[DOCS] doc 383: bring sql datasource configuration examples under test

Rachel-Reverie opened the pull request:
Changes proposed in this pull request:
- Create scripts to contain code snippet examples for the "How to configure a SQL Datasource" guide and puts them under test.

### Definition of Done

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have run any local integration tests and made sure that nothing is broken.


Thank you for submitting!

Rachel-Reverie committed:
DOC-383
- Adds the "source of truth" full SQL Datasource configurations

Rachel-Reverie committed:
DOC-383
- Adds the IntegrationTestFixture for the SQL example script to test_script_runner.py

Rachel-Reverie committed:
DOC-383
- Adds example script snippets and validations for SQL Datasource configurations

Rachel-Reverie committed:
DOC-383
- Adds an additional "Full" configuration, due to the SQL examples combining the single-batch and multi-batch assets under different names after the step detailing how to simply define "an asset".
- Edits to some of the configurations for consistency.

Rachel-Reverie committed:
Merge branch 'develop' into d/docs/doc-383/bring_sql_datasource_configuration_examples_under_test

Rachel-Reverie committed:
DOC-383
- Tests all full configurations using `test_yaml_config` and a sqlite database containing all months of 2020 taxidata in a single table.

ghost commented:
<details>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=6466&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

  - Make big changes easier — review code in small groups of related files
  - Know where to start — see the whole change at a glance
  - Take a code tour — explore the change with an interactive tour
  - Make comments and review — all fully sync’ed with github

    [Try it now!](https://app.codesee.io/r/reviews?pr=6466&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)
</details>

[![Review these changes using an interactive CodeSee Map](https://s3.us-east-2.amazonaws.com/maps.codesee.io/images/github/great-expectations/great_expectations/6466/0ca70adc/459bbcb8ee201bcdf4b9b01b993b651e21b7a37b.svg)](https://app.codesee.io/r/reviews?pr=6466&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
![CodeSee Map Legend](https://app.codesee.io/Legend_v2.png)

Rachel-Reverie committed:
DOC-383
- Adds and tests the runtime datasource and configured datasource configurations for a SqlAlchemyExecutionEngine.

Rachel-Reverie committed:
DOC-383
- Updates all of the code blocks to reference code examples that are under test.

Rachel-Reverie committed:
Merge branch 'develop' into d/docs/doc-383/bring_sql_datasource_configuration_examples_under_test

austiezr commented on docs/guides/connecting_to_your_data/datasource_configuration/sql_components/_tab_data_connector_example_configurations_runtime.mdx:
````diff
@@ -33,29 +19,12 @@ datasource_config: dict = {
-        }
-    }
-}
+```python name="configured sql datasource configuration with data_connector class_name defined"
````
These blocks are all referencing a `ConfiguredAssetSqlDataConnector` rather than a `RuntimeDataConnector`

austiezr reviewed (changes requested):
One note, otherwise 🚀 !

austiezr commented on docs/guides/connecting_to_your_data/datasource_configuration/sql_components/_tab_data_connector_example_configurations_runtime.mdx:
````diff
@@ -9,21 +9,7 @@ import TipCustomDataConnectorModuleName from '../components/_tip_custom_data_con
-        }
-    }
-}
+```python name="sql datasource configuration with empty configured sql data_connector"
````
These blocks are all referencing a ConfiguredAssetSqlDataConnector rather than a RuntimeDataConnector

Rachel-Reverie replied:
Should be resolved now.

austiezr commented on docs/guides/connecting_to_your_data/datasource_configuration/sql_components/_tab_data_connector_example_configurations_runtime.mdx:
````diff
@@ -33,29 +19,12 @@ datasource_config: dict = {
 
-```python
-    "class_name": "RuntimeDataConnector",
+```python name="define data_connector class_name for configured sql datasource"
````
These blocks are all referencing a ConfiguredAssetSqlDataConnector rather than a RuntimeDataConnector

Rachel-Reverie replied:
Should be resolved now.

Rachel-Reverie committed:
DOC-383
- Correction to runtime examples.

Rachel-Reverie committed:
Merge remote-tracking branch 'origin/d/docs/doc-383/bring_sql_datasource_configuration_examples_under_test' into d/docs/doc-383/bring_sql_datasource_configuration_examples_under_test

austiezr reviewed (approved):
🚀

austiezr committed:
Merge branch 'develop' into d/docs/doc-383/bring_sql_datasource_configuration_examples_under_test

github-actions[bot] committed:
Merge develop into d/docs/doc-383/bring_sql_datasource_configuration_examples_under_test

github-actions[bot] committed:
Merge develop into d/docs/doc-383/bring_sql_datasource_configuration_examples_under_test
